### ISSUE TITLE
Uploading file crashes application in node v14 

### ISSUE THREAD

**Issue by mohsin51** [jaydenseric/graphql-upload#218]
Tried node versions 14.4.0 and 14.13.0

Error details
`TypeError: isObject is not a function
    at Busboy.<anonymous> (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:77533:18)
    at Busboy.emit (events.js:315:20)
    at Busboy.emit (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:1680306:33)
    at PartStream.onEnd (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:1683101:15)
    at PartStream.emit (events.js:327:22)
    at Dicer.onPart (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:1682959:13)
    at Dicer.emit (events.js:315:20)
    at Dicer.emit (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:1683249:35)
    at Dicer._oninfo (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:1683350:12)
    at SBMH.<anonymous> (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:1683296:10)
    at SBMH.emit (events.js:315:20)
    at SBMH._sbmh_feed (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:76334:14)
    at SBMH.push (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:76231:14)
    at Dicer._write (/home/in01-nbk-741/Work/apps/intech-apps/dist/apps/platform/backend/main.js:1683278:17)
    at writeOrBuffer (_stream_writable.js:352:12)
    at Dicer.Writable.write (_stream_writable.js:303:10)
`

This happens when you build your application in production but in dev mode it's working
This is my implementation
``` makePrismaSchema({
  types: [
    queries,
    mutations,
    upload, //  imported as { GraphQLUpload } from 'graphql-upload';
     .......
     ....
  ]
});


const server = new ApolloServer({
  playground: true,
  introspection: true,
  tracing: true,
  uploads: false, // disable built-in uploads of apollo
  schema: schemaWithMiddleware,
  context: { prisma, client },
  plugins: [
    loggingPlugin,
  ],
});

app.use(graphqlUploadExpress({ maxFileSize: 10000000 })); // 10 mb
server.applyMiddleware({
  app,
  path: '/${API_URL}',
  onHealthCheck: () => new Promise((resolve, reject) => {
    axios(NEXUS_HEALTH_CHECK_URL).then(res => {
      resolve(res);
    }).catch(e => reject(e));
  }),
});

//mutation resolver
 resolve: async (parent, { file }) => {
      let models = [];
      const {
        createReadStream, filename, mimetype, encoding,
      } = await file;
 .....
```

linked issues : [issue#1](https://github.com/apollographql/apollo-server/issues/3508) [issue#2](https://github.com/jaydenseric/graphql-upload/issues/170)

Initially i was using built-in upload in apollo server which was causing maximum call stack size error
after doing modification according to this [comment](https://github.com/apollographql/apollo-server/issues/3508#issuecomment-662371289). it was working fine in dev mode but crashing when bundled for production.

**Comment by jaydenseric** [comment]
I'm sorry to hear you have been having a hard time dealing with Apollo Server using a very outdated dependency; it makes me sad because I've put in a lot of effort to make file uploads elegant and easy - all the `graphql-upload` tests are passing in Node.js v14:

https://github.com/jaydenseric/graphql-upload/runs/705151521

I'm not in a good position to help you figure out bundling workarounds for Apollo packages, but best of luck to you!

Instead of publishing semver major releases when necessary, the timing of Apollo Server and Apollo Client major version releases are dictated by their marketing plans. This can result in years between major releases, resulting in a backlog of critical dependency updates at any given time. Additionally Apollo has a small engineering team that is failing to keep up with basic maintenance of their diverse and numerous packages, e.g.

https://github.com/apollographql/subscriptions-transport-ws/issues/777#issuecomment-702908551

I wish the Apollo team well and hope they succeed, but my instinct is that such problems arising from a marketing focus, organizational dysfunction, engineering bloat, and the engineer shortage won't get better with time. A solution is to invest in a more elegant stack that is not at the mercy of a single for-profit company. Here is mine that I have built from scratch and aggressively maintain:

- Server: https://github.com/jaydenseric/graphql-api-koa
- Client: https://github.com/jaydenseric/graphql-react

I'm not the only person thinking this way, I've noticed a lot of diversification away from Apollo recently. GraphQL is an idea bigger than one company, and the [GraphQL foundation](https://foundation.graphql.org) is doing good work on that front.

**Comment by TheophileSketo** [comment]
The problem is [this](https://github.com/jaydenseric/graphql-upload/blob/0f7882d8867b33019c0e1e5e3c485d05ec8114cb/public/processRequest.js#L6) import. It seems that isobject at version 4.0.0 ships an ES module with a default export which after processing with webpack results in a broken import. If you can modify your webpack config changing resolve.mainFields to ['main', 'module'] should workaround the issue.

**Comment by jaydenseric** [comment]
@TheophileSketo I'm not following the problem you're having, or why you would suggest that as a solution. The outcome is basically the same if webpack resolves the `isobject` v4.0.0 `main` or `module` entrypoint:

- https://unpkg.com/browse/isobject@4.0.0/index.cjs.js
- https://unpkg.com/browse/isobject@4.0.0/index.js

`module.exports =` is similar to `export default`; modern webpack and Node.js should both be able to resolve `require('isobject')` just fine.

That said, I would do things differently that how they have (e.g. an ESM file should have a `.mjs` extension, use `package.json` `exports` field instead of `module`, etc.). Really it's an anti-pattern to ship the same code in two seperate ESM and CJS files as that can introduce the dual-package hazard, and there is no tree shaking benefits to using an ESM file when the whole package exports only a single function anyway.

**Comment by TheophileSketo** [comment]
The problem is webpack creates this export `__webpack_require__.d(__webpack_exports__, "default", function() { return isObject; });` and this import `const isObject = __webpack_require__(1195);` when it imports the esm file. Although it might be that typescript is messing it up before webpack.
